### ISSUE TITLE
DISTINCT applies to all bound variables

### ISSUE THREAD

**Issue by vhf** [zazuko/query-rdf-data-cube#23]
> why is `.distinct` a method of the dimension even? Shouldn't that rather be:
> 
> ```js
> cube.query()
>   .select({foo: fooDimension})
>   .distinct()
>   .execute()
> ```
> 
> (I think SPARQL `DISTINCT` applies to all bound variables not just single ones?)

_Originally posted by @herrstucki in https://github.com/zazuko/query-rdf-data-cube/issues/20#issuecomment-530819013_

**Comment by vhf** [comment]
Fixed: 43a813be54c7ea5ab0a89a6b757a2469ec237355

Note that `.distinct()` is still a method on Components because it can be useful there, eg.

```js
const query = dataCube.query().select({
      foo: fooMeasure.avg().distinct(),
    })
```

**Comment by vhf** [comment]
Published as `@zazuko/query-rdf-data-cube@0.0.11`
